convert_image.go

```diff
@@ -45,5 +44,9 @@ func convertImage(r io.Reader) ([]byte, error) {
 	// Re-encode as low quality jpeg
 	var buf bytes.Buffer
 	err = jpeg.Encode(&buf, img, &jpeg.Options{Quality: 40})
-	return buf.Bytes(), err
+	converted := buf.Bytes()
+	if len(converted) < len(orig) {
+		return converted, err
+	}
+	return orig, err
```

Comment:
I try my best to follow the [Return Early pattern](https://medium.com/swlh/return-early-pattern-3d18a41bba8) in this project, as it is preferred in most Go codebases.

One of the key points of this pattern is the [Happy Path](https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88), which I don't think is followed here.

Following that pattern plus the convenient `err := doStuff(); err != nil` syntax and using `Buffer.Len()`, I get this code:

```go
if err := jpeg.Encode(&buf, img, &jpeg.Options{Quality: 40}); err != nil {
	return orig, err
}
converted := buf.Bytes()
if buf.Len() >= len(orig) {
	return orig, nil
}
return buf.Bytes(), nil
```

I think that's more readable as we here explicitly handle the error right away by returning early, and can clearly see the happy path and exception handling by following the indent level.

I'd prefer if you change that please 🙂 

PS: You can go ahead and squash the commits afterwards, and I'll merge it asap. Thanks!

Reply:
Done and squashed